### PR TITLE
Make possibility to get EntityPersister.php. Refactor the code

### PR DESCRIPTION
**Pull request by dzianis-hrynko**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
         "email": "php@graphaware.com"
     }],
     "require": {
-        "php": "^8.0",
+        "php": ">8.0",
```
Comment (gep) [inline composer.json]: `>=8.0`

Comment (dzianis-hrynko) [reply composer.json]: fixed

File: src/Proxy/ProxyFactory.php [hunk 2]
```diff
@@ -197,11 +197,11 @@ private function getInitializerFor(RelationshipMetadata $relationship)
     {
```
Comment (KirylPHP) [inline src/Proxy/ProxyFactory.php]: Return type declaration SingleNodeInitializer | NodeCollectionInitializer was missed

File: src/UnitOfWork.php [hunk 3]
```diff
@@ -888,7 +868,8 @@ private function updateNodes(): void
         $statements = [];
         foreach ($this->nodesScheduledForUpdate as $entity) {
             $this->traverseRelationshipEntities($entity);
-            $statements[] = $this->getPersister($entity::class)->getUpdateQuery($entity);
+            $statements[] = $this->entityManager->getEntityPersister($entity::class)->getUpdateQuery($entity);
+            $statements[] = $this->entityManager->getEntityPersister($entity::class)->getUpdateQuery($entity);
```
Comment (KirylPHP) [inline src/UnitOfWork.php]: Can you please clarify purpose of this duplicate?

File: src/UnitOfWork.php [hunk 4]
```diff
@@ -259,7 +252,7 @@ public function detectRelationshipEntityChanges()
         }
     }
 
-    public function addManagedRelationshipEntity($entity, $pointOfView, $field)
+    public function addManagedRelationshipEntity($entity)
```
Comment (KirylPHP) [inline src/UnitOfWork.php]: Return type declaration 'void' was missed

Comment (KirylPHP) [reply src/UnitOfWork.php]: Type declaration 'object' before $entity was missed

File: src/UnitOfWork.php [hunk 5]
```diff
@@ -449,16 +439,6 @@ public function getEntityById(int $id): ?object
...
-    {
-        if (!array_key_exists($class, $this->persisters)) {
-            $classMetadata = $this->entityManager->getClassMetadataFor($class);
-            $this->persisters[$class] = new EntityPersister($this->entityManager, $class, $classMetadata);
-        }
-
-        return $this->persisters[$class];
-    }
-
     public function getRelationshipEntityPersister($class): RelationshipEntityPersister
```
Comment (KirylPHP) [inline src/UnitOfWork.php]: Type declaration 'string' before $class was missed

File: src/UnitOfWork.php (hunk 3 above)
Comment (dzianis-hrynko) [reply src/UnitOfWork.php]: fixed

File: src/Proxy/ProxyFactory.php (hunk 2 above)
Comment (dzianis-hrynko) [reply src/Proxy/ProxyFactory.php]: fix

File: src/UnitOfWork.php (hunk 5 above)
Comment (dzianis-hrynko) [reply src/UnitOfWork.php]: fix

File: src/UnitOfWork.php (hunk 4 above)
Comment (dzianis-hrynko) [reply src/UnitOfWork.php]: fix

**Review by petergodmanap — APPROVED** [review]

**Pull request merged**